src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,191 @@
+        if (tasks.length <= 1) {
+            throw new DukeException("OOPS!!! Please tell me what you want to find!!");
+        }
+    }
+
+    /**
+     * This method when called reformats the date input by the user to a Date object.
+     *
+     * @param date The date string to be reformatted.
+     * @return The <code>Date</code> in a formatted manner.
+     * @throws DukeException If there is a problem with data processing, loading or saving.
+     */
+    public static Date dateFormatter(String date) throws DukeException {
+        try {
+            SimpleDateFormat formatter = new SimpleDateFormat("dd/MM/yyyy HHmm");
+            Date parseDate = formatter.parse(date);
+            return parseDate;
+        } catch (ParseException e) {
+            throw new DukeException("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    /**
+     * This method when called will result in the returning of a <code>Command</code>
+     * which can be executed to result in either a deletion of the task, an addition of
+     * a task or to mark a task as done from the task list.
+     *
+     * @param command The user's input string.
+     * @return A class of command depending on the type of user input.
+     * @throws DukeException If there is a problem with data processing, loading or saving.
+     */
+    public static Command parse(String command) throws DukeException {
+        if (command.equals("bye")) {
+            return new ExitCommand();
+        } else if (command.equals("list")) {
+            return new ListCommand();
+        } else {
+            String[] task = command.split(" ");
+            String instruction = task[0];
+
+            if (instruction.equals("done")) {
+                int taskNumber = Integer.parseInt(task[1]);
+                return new DoneCommand(taskNumber - 1);
+            } else if (instruction.equals("delete")) {
+                try {
+                    int taskNumber = Integer.parseInt(task[1]);
+                    return new DeleteCommand(taskNumber - 1);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    throw new DukeException("OOPS!!! Provide a task number to delete!");
+                }
+            } else {
+                switch (instruction) {
+                case "todo": {
+                    try {
+                        todoCheck(task);
+                        Task todo = new Todo(command.substring(5));
+                        return new AddCommand(todo);
+                    } catch (DukeException e) {
+                        System.err.println("Something went wrong: " + e.getMessage());
+                        break;
+                    }
+                }
+                case "deadline": {
+                    try {
+                        deadlineCheck(task, command);
+                        Task deadline = new Deadline(command.substring(9, command.indexOf("/by") - 1),
```

Comment:
Same issue as the code for todo.
Would it work if the user enters "deadline2103T iP /by 20/10/2019 2000"?